xla/service/gpu/tests/gemm_rewrite_test.cc

```diff
@@ -3719,6 +3719,167 @@ ENTRY test {
       )");
 }
 
+TEST_F(CublasLtGemmRewriteTest, ReluActivationWithAux) {
```

Comment:
In most real-world ML models, there is a path from the forward matmul to the backward matmul, since you need intermediate activation in the forward pass to compute the backward pass. This affects the order GemmRewriterVisitor processes nodes, which could affect correctness, so would be good to test this case.

Can you add an FP8 relu aux test that tests this, by passing the output of the first dot as an input to the second dot?

Reply:
There is a [FP8 test](https://github.com/openxla/xla/pull/10977/files#diff-71bce65d79d6395cde36b49b4504c6b0c73694cee153e5685a8b46ff146f09e0R5595) for this purpose.